src/Features/Core/Portable/EditAndContinue/TraceLog.cs

```diff
@@ -90,7 +90,20 @@ public void DebugWrite(string format, params Arg[] args)
             Debug.WriteLine(entry.ToString(), _id);
         }
 
-        // test only
-        internal Entry[] GetEntries() => _log;
+        internal TestAccessor GetTestAccessor()
+            => new TestAccessor(this);
+
+        internal readonly struct TestAccessor
+        {
+            private readonly TraceLog _traceLog;
+
+            public TestAccessor(TraceLog traceLog)
+            {
+                _traceLog = traceLog;
+            }
+
+            internal ref readonly Entry[] Entries
+                => ref _traceLog._log;
```

Comment:
feels odd... why does this need to be a ref-readonly to an array?  why not have a prop that returns a direct ref to the array?

Reply:
I'll remove the ref type here since the field is readonly.

Reply:
WFM.  I tend to like only seeing 'ref' for very sensitive cases where it really is a critical part of things.  Otherwise, i think it just adds cognitive overhead.  That said, it's probably because it's just so new.  If you honestly prefer it in cases go right ahead :)

Reply:
One of the goals of the test accessor is to provide direct access to the underlying item which is protected. For example, it should be possible to use `Interlocked.Exchange` or `Volatile.Read` with the test accessor.